File: 189-rotate-array/rotate-array.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {void} Do not return anything, modify nums in-place instead.
 */
var rotate = function(nums, k) {
    let n = nums.length
    k = k % n
    let temp = []

    for(let i = n-k; i <= n; i++){
        temp.push(nums[i])
    }

    for (let i = n - 1; i >= k; i--) {
      nums[i] = nums[i - k];
     
    }
    

    
    for (let i = 0; i < k; i++) {
        nums[i] = temp[i];
        }
};

File: 189-rotate-array/rotate-array.java
class Solution {
  public void rotate(int[] nums, int k) {
    int n = nums.length;
    k = k % n;
    ArrayList<Integer> list = new ArrayList<>();

    for (int i = n - k; i < n; i++) {
      list.add(nums[i]);
    }

    for (int i = n - 1; i >= k; i--) {
      nums[i] = nums[i - k];
    }

    for (int i = 0; i < k; i++) {
      nums[i] = list.get(i);
    }

  }
}